service/proxy/src/main/java/org/kaazing/gateway/service/proxy/ProxyServiceExtensionSpi.java

```diff
@@ -34,5 +36,32 @@
      * @param acceptSession - the accept side of the end-to-end proxy connection
      * @param connectSession - the connect side of the end-to-end proxy connection
      */
-    public abstract void proxiedConnectionEstablished(IoSessionEx acceptSession, IoSessionEx connectSession);
+    default void proxiedConnectionEstablished(IoSessionEx acceptSession, IoSessionEx connectSession) {
+        // no behavior by default so an extension can decide where it attaches its behavior
+    }
+
+    /**
+     * Whenever a new client connection is received (on the configured "accept")
+     * the extension is giving an opportunity to initialize the associated session.
+     *
+     * @param acceptSession - the accept side of the end-to-end proxy connection
+     * @param properties - the service properties that contain the values used to
+     *                     configure the proxy service
+     */
+    default void initAcceptSession(IoSession acceptSession, ServiceProperties properties) {
```

Comment:
At some point, we will remove ServiceProperties. For now, we will continue with this changeset.